api/pipeline_spec.proto

```diff
@@ -0,0 +1,420 @@
+syntax = "proto3";
+
+package ml_pipelines;
+
+import "google/protobuf/any.proto";
+
+// The spec of a pipeline.
+message PipelineSpec {
+  // The metadata of the pipeline.
+  PipelineInfo pipeline_info = 1;
+
+  // A list of pipeline tasks, which form a DAG.
+  repeated PipelineTaskSpec tasks = 2;

```

Comment:
Can you please show an example with sub-DAGs?

Reply:
Sub-DAG is a future feature on top of the schema. It won't be supported in this PR.